pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Subscription.java

```diff
@@ -79,6 +80,8 @@ default long getNumberOfEntriesDelayed() {
 
     CompletableFuture<Entry> peekNthMessage(int messagePosition);
 
+    CompletableFuture<Entry> getMessageById(long ledgerId, long entryId);
```

Comment:
If we can use ManagedLedger directly, this method also can be removed.

Reply:
Good suggestion. Fixed them in the following update.